github-backup.py

```diff
@@ -27,118 +42,439 @@ def main():
+        }
+
+    if args.account:
+        process_account(gh, account, args)
+
+    if args.include_gists:
+        for gist in account.get_gists():
+            RepositoryBackup(gist, args).backup()
+
+    if args.include_starred_gists and hasattr(account, 'get_starred_gists'):
+        for gist in account.get_starred_gists():
+            RepositoryBackup(gist, args).backup()
+
+    if not args.skip_repos:
+        repos = account.get_repos(**filters)
+        for repo in repos:
+            if args.skip_forks and repo.fork:
+                continue
+
+            RepositoryBackup(repo, args).backup()
 
 def init_parser():
     """Set up the argument parser."""
 
-    parser = ArgumentParser(description="makes a backup of all of a github user's repositories")
+    parser = ArgumentParser(description="makes a backup of a github user's account")
 
-    parser.add_argument("login_or_token", help="A Github username or token")
+    parser.add_argument("login_or_token", help="A Github username or token for authenticating")
     parser.add_argument("backupdir", help="The folder where you want your backups to go")
     parser.add_argument("-v", "--visibility", help="Filter repos by their visibility", choices=['all', 'public', 'private'], default='all')
     parser.add_argument("-a", "--affiliation", help="Filter repos by their affiliation", action='append', type=str, default=['owner'], choices=['owner', 'collaborator', 'organization_member'])
     parser.add_argument("-d", "--debug", help="Show debug info", action="store_true")
     parser.add_argument("-q", "--quiet", help="Only show errors", action="store_true")
     parser.add_argument("-m", "--mirror", help="Create a bare mirror", action="store_true")
     parser.add_argument("-f", "--skip-forks", help="Skip forks", action="store_true")
+    parser.add_argument("--skip-repos", help="Skip backing up repositories", action="store_true")
     parser.add_argument("-g", "--git", nargs="+", help="Pass extra arguments to git", type=list, default=[], metavar="ARGS")
     parser.add_argument("-t", "--type", help="Select the protocol for cloning", choices=['git', 'http', 'ssh'], default='ssh')
     parser.add_argument("-s", "--suffix", help="Add suffix to repository directory names", default="")
-    parser.add_argument("-p", "--password", help="Authenticate with Github API")
+    parser.add_argument("-u", "--username", help="Backup USER account", metavar="USER")
+    parser.add_argument("-p", "--password", help="Authenticate with Github API (give no argument to check ~/.github-backup.conf or prompt for a password)", nargs="?", default=False)
     parser.add_argument("-P", "--prefix", help="Add prefix to repository directory names", default="")
     parser.add_argument("-o", "--organization", help="Backup Organizational repositories", metavar="ORG")
+    parser.add_argument("-A", "--account", help="Backup account data", action='store_true')
+    parser.add_argument('--all',
```

Comment:
Can we add a short option `-a` here as well?

Reply:
The short option `-a` is already used for `--affiliation`.  This changes the semantics of the options and might mess with people's scripts, so I'm wary of changing this.